src/Features/Blockcore.Features.BlockStore/BlockStoreFeature.cs

```diff
@@ -210,12 +215,28 @@ public static IFullNodeBuilder UseBlockStore(this IFullNodeBuilder fullNodeBuild
                         services.AddSingleton<IBlockStoreQueueFlushCondition, BlockStoreQueueFlushCondition>();
                         services.AddSingleton<IAddressIndexer, AddressIndexer>();
 
-                        services.AddSingleton<IPrunedBlockRepository, PrunedBlockRepository>();
                         services.AddSingleton<IPruneBlockStoreService, PruneBlockStoreService>();
                     });
             });
 
             return fullNodeBuilder;
         }
+
+        private static void AddDbImplementation(IServiceCollection services, NodeSettings settings)
```

Comment:
DRY, possible to reuse this method across features?

Reply:
no each feature has its own implementation of a repo (store has store repo coins has coinddb etc...)